### DISCUSSION TITLE
Dolphins Unprotect Protected Drops

### DISCUSSION THREAD

**Discussion by dvdcd** [GriefPrevention/GriefPrevention#700] [General]
### What happened:
Protected drops become unprotected when played with by a dolphin

### What was expected:
Drops stay protected/are unable to be picked up by dolphins until the player uses /unlockdrops

### Steps to reproduce:
1. Die with protected drops on
2. Have dolphin pick up drops
3. Alternate accounts can now pick up players drops

### Server and GriefPrevention version:
___This can be found by running `/version` and `/version GriefPrevention` on your server.___
Paste of `/version`: 
This server is running Paper version git-Paper-58 (MC: 1.15.1) (Implementing API version 1.15.1-R0.1-SNAPSHOT)
Paste of `/version GriefPrevention`: 
 GriefPrevention version 16.12.0-d4d7ac4

### Debugging steps attempted

- [x] I attempted running only GriefPrevention on the server
  - [x] Issue still occurs even if GriefPrevention is the only plugin running
- [x] I attempted testing for the issue on a new server
  - [x] Issue still occurs even after testing on a new server

**Comment by RoboMWM** [comment]
I didn't know dolphins can pickup items. Not really sure what to do here other than prevent dolphins from picking up said items - as there's no way to tag the itemstack itself unless the item's lore or NBT is modified.

**Comment by Jikoo** [comment]
The [PlayerPickupItemEvent](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/player/PlayerPickupItemEvent.html) is now deprecated in favor of the EntityPickupItemEvent. Not sure why they didn't just change it to be more like the other player events that directly inherit from entity events, but I'm sure at some point Spigot will start pushing plugins to listen to that event instead. Could certainly add this, but it would be nice to still allow hostiles to pick up player gear to make retrieval more challenging.
